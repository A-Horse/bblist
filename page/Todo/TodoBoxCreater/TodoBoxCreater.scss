@import 'style/variable';


.todo-box-creater--toggle {
  cursor: pointer;
  padding: 6px 20px;
  color: #555;
  font-size: 1.1rem;

  i {
    vertical-align: middle;
    margin-right: 10px;
    margin-top: -1px;
  }

  @media screen and (max-width: 600px) {
    padding: 3px 0px 2px 12px;
    color: $white;
    font-size: 1.2rem;

    i {
      margin-top: 0;
    }
  }
}

.todo-box-creater-modal {
  top: 20%;
  padding: 30px;

  .todo-box-creater--heading {
    color: #555;
    font-weight: 500;
    font-size: 1.3rem;
    margin-bottom: 20px;
  }

  .todo-box-name--input {
    border-radius: 3px;
    margin-right: 10px;
    height: 30px;
    border: 1px solid #e8e8e8;
    padding: 0 5px;
  }
}
