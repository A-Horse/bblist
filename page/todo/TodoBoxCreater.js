import React, {Component} from 'react';
import {connect} from 'react-redux';

import {addBodyEventListenerOnce} from 'actions/event/body';
import {createTodo, getTodoList} from 'actions/todo/todos';
import {AddIcon, AlarmIcon, CloseIcon} from 'services/svg-icons';
import {Button} from 'components/widget/Button';
import DatePicker from 'components/date-picker/DatePicker';
import Popup from 'components/Popup';
import {Select} from 'components/widget/Select';
import Textarea from 'react-textarea-autosize';
import {Modal} from 'components/widget/Modal';
import Input from 'components/widget/Input';

import 'style/page/todo/todo-box-creater.scss';

class TodoBoxCreater extends Component {
  componentWillMount() {
    this.state = {toggle: false};
  }

  toggle() {
    this.setState({toggle: true});
  }

  render() {
    return (
      <div className='todo-box-creater--toggle' onClick={this.toggle.bind(this)}>
        <AddIcon className='add-icon' />
        <span className='toggle-text'>Create Todo Box</span>
      </div>
    );
  }

  close() {
    this.setState({toggle: false});
  }

  createTodoBox() {

  }

  renderCreater() {
    return (
      <Modal className='todo-box-creater' toggle={this.state.toggle} close={::this.close}>
        <CloseIcon className='clear-icon' onClick={::this.closeModal}/>

        <div className='todo-box-creater--heading'>Create Todo Box:</div>

        <Input className='todo-box-name--input' type='text' ref='name' placeholder='Board Name'/>
        <Button styleType='primary' className='taskboard-creater--create-button' onClick={::this.createTodoBox} >OK</Button>
      </Modal>
    );
  }

}

export default TodoBoxCreater;
