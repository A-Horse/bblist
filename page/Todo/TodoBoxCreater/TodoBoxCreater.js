import React, { Component } from 'react';
import { connect } from 'react-redux';

import { addBodyEventListenerOnce } from 'actions/event/body';
import { createTodo, getTodoList } from 'actions/todo/todos';
import { AddIcon, AlarmIcon, CloseIcon } from 'services/svg-icons';
import { Button } from 'components/widget/Button';
import DatePicker from 'components/DatePicker/DatePicker';
import Popup from 'components/Popup';
import { Select } from 'components/widget/Select';
import Textarea from 'react-textarea-autosize';
import { Modal } from 'components/widget/Modal/Modal';
import Input from 'components/widget/Input';

import './TodoBoxCreater.scss';

class TodoBoxCreater extends Component {
  constructor() {
    super();
    this.close = this.close.bind(this);
  }
  componentWillMount() {
    this.state = { toggle: false };
  }

  toggle() {
    this.setState({ toggle: true });
  }

  render() {
    return (
      <div className="todo-box-creater--toggle" onClick={this.toggle.bind(this)}>
        <AddIcon className="add-icon" />
        <span className="toggle-text">Create Todo Box</span>
      </div>
    );
  }

  close() {
    this.setState({ toggle: false });
  }

  renderCreater() {
    return (
      <Modal className="todo-box-creater" toggle={this.state.toggle} close={this.close}>
        <CloseIcon className="clear-icon" onClick={this.close} />

        <div className="todo-box-creater--heading">Create Todo Box:</div>

        <Input className="todo-box-name--input" type="text" ref="name" placeholder="Board Name" />
        <Button styleType="primary" className="taskboard-creater--create-button">
          OK
        </Button>
      </Modal>
    );
  }
}

export default TodoBoxCreater;
