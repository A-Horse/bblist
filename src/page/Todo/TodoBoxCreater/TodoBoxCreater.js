//
import React, { Component } from 'react';
import { Input, Form, Button, Icon, Modal } from 'antd';
import { formShape } from 'rc-form';

import './TodoBoxCreater.less';

const FormItem = Form.Item;

export class TodoBoxCreaterForm extends Component {
  state = { toggle: false, name: '' };

  close = () => {
    this.setState({ toggle: false });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.form.validateFieldsAndScroll((err, values) => {
      if (!err) {
        this.props.actions.ADD_TODOBOX_REQUEST(values);
        this.handleCancel();
      }
    });
  };

  handleCancel = () => {
    this.setState({
      toggle: false
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div>
        <Button
          className=""
          onClick={event => {
            event.stopPropagation();
            this.setState({ toggle: true });
          }}
        >
          <Icon type="folder-add" />
          Add Todo Box
        </Button>

        <Modal
          className="todo-box-creater-modal"
          title="Create Todo Box:"
          onCancel={this.handleCancel}
          visible={this.state.toggle}
          footer={null}
        >
          <Form onSubmit={this.handleSubmit}>
            <FormItem>
              {getFieldDecorator('name', {
                rules: [{ required: true, message: 'Please input Todo Box name' }]
              })(<Input type="text" placeholder="Todo Box" />)}
            </FormItem>

            <FormItem>
              <Button type="primary" htmlType="submit">
                Done
              </Button>
            </FormItem>
          </Form>
        </Modal>
      </div>
    );
  }
}

export const TodoBoxCreater = Form.create()(TodoBoxCreaterForm);
