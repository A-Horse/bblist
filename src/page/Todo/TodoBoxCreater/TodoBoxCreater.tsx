import './TodoBoxCreater.scss';

import {  Form, Icon, Input, Modal } from 'antd';
import React, { Component } from 'react';
import { AppButton } from '../../../components/widget/Button';

const FormItem = Form.Item;

export class TodoBoxCreaterForm extends Component<any> {
  state = { toggle: false, name: '' };

  close = () => {
    this.setState({ toggle: false });
  };

  handleSubmit = (event: any) => {
    event.preventDefault();
    this.props.form.validateFieldsAndScroll((err: any, values: any) => {
      if (!err) {
        this.props.actions.ADD_TODOBOX_REQUEST(values);
        this.handleCancel();
      }
    });
  };

  handleCancel = () => {
    this.setState({
      toggle: false
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <div onClick={(event: any) => {
        event.stopPropagation();
        this.setState({ toggle: true });
      }}>
        <Icon type="folder-add" />
          Add Todo Box

        <Modal
          className="todo-box-creater-modal"
          title="Create Todo Box:"
          onCancel={this.handleCancel}
          visible={this.state.toggle}
          footer={null}
        >
          <Form onSubmit={this.handleSubmit}>
            <FormItem>
              {getFieldDecorator('name', {
                rules: [{ required: true, message: 'Please input Todo Box name' }]
              })(<Input type="text" placeholder="Todo Box" />)}
            </FormItem>

            <FormItem>
              <AppButton type="submit">
                Done
              </AppButton>
            </FormItem>
          </Form>
        </Modal>
      </div>
    );
  }
}

export const TodoBoxCreater = Form.create()(TodoBoxCreaterForm);
