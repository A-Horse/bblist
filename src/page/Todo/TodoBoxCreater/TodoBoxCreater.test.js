import React from "react";
import TodoBoxCreater from "./TodoBoxCreater";
import renderer from "react-test-renderer";

test("TodoBoxCreater", () => {
  const actions = {};
  const tree = renderer.create(<TodoBoxCreater actions={actions} />);
  expect(tree).toMatchSnapshot();
});
