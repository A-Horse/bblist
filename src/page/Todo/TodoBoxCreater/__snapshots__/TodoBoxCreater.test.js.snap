// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoBoxCreater 1`] = `
<div>
  <div
    className="todo-box-creater--toggle"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-plus"
    />
    <span
      className="toggle-text"
    >
      Create Todo Box
    </span>
  </div>
  <noscript />
</div>
`;
